.github/workflows/ci.yml

```diff
@@ -19,13 +19,10 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
           allow-prereleases: true
+      - run: ${{ ( startsWith( matrix.os, 'windows' ) && 'bash add_libmagic.sh' ) || 'sudo -E bash add_libmagic.sh' }}
```

Comment:
It's a strange command, would it be required from the user to use it on each installation?

Reply:
```suggestion
      - if: runner.os == 'Windows'
        shell: bash
        run: add_libmagic.sh || sudo -E bash add_libmagic.sh
```

Reply:
> Each _**Windows**_ installation.

Would it be possible to make this command a part of the setup? Isn't the main idea of this PR to make installation of binaries as a part of the PIP package?

Why would admin privileges be required? Isn't it possible to download the DLL in the same folder where the package is? If not, would it be possible to ship binaries together?

As an alternative solution, we can require package with binaries as a sub dependency.

Reply:
`add_libmagic.sh`, introduced in this PR, adds the assets to the current folder which need to be packaged into the wheel. since `ci.yml` is installing from source (and not from wheel), this command needs to run here, regardless of platform (cc @cclauss)

essentially you're seeing an install from source here. install from wheel won't be needing this command. and if you already have libmagic installed yourself, you also won't be needing this command to install from source.

all in all, this shell script is only a helper for github actions, and won't be needed by PyPI users.

I hope that answers all your questions!

Reply:
```suggestion
      - name: Copy libmagic into magic dir
        run: ${{ ( startsWith( matrix.os, 'windows' ) && 'bash add_libmagic.sh' ) || 'sudo -E bash add_libmagic.sh' }}
```